db/errors.go

```diff
@@ -0,0 +1,55 @@
+package db
+
+import (
+	"errors"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// HandlePGError processes PostgreSQL errors and logs them appropriately
+func HandlePGError(err error, logger *zap.SugaredLogger, operation string) error {
```

Comment:
Oof, also jetzt hast du die Mühe schon reingesteckt und ich würde es so lassen, aber ich kein Fan davon. Die generischen Errors gerne, aber alles was spezifisch ist, sollte nicht in so einer `can-do-it-all` Funktion abgefangen werden.

Reply:
Bin auch kein Fan, zeit ist aber knapp und hab kein bock das überall anzupassen ^^